plugins/modules/instance.py

```diff
@@ -384,6 +418,33 @@ def get_ssh_key_ids(self):
 
         return ssh_key_ids
 
+    def get_vpc_ids(self):
+        vpc_names = list(self.module.params["vpcs"])
+        vpcs = self.query_list(path="/vpcs", result_key="vpcs")
+
+        vpc_ids = list()
+        for vpc in vpcs:
+            if vpc["description"] in vpc_names:
+                vpc_ids.append(vpc["id"])
+                vpc_names.remove(vpc["description"])
+
+        if vpc_names:
+            self.module.fail_json(msg="VPCs not found: %s" % ", ".join(vpc_names))
+
+        return vpc_ids
+
+    def get_instance_vpcs(self, resource):
+        path = "/instances/%s/vpcs" % resource["id"]
+        vpcs = self.query_list(path=path, result_key="vpcs")
+
+        # Workaround to get the description field into the list
+        result = list()
+        for vpc in vpcs:
+            vpc_detail = self.query_by_id(resource_id=vpc["id"], path="/vpcs", result_key="vpc")
+            vpc["description"] = vpc_detail["description"]
+            result.append(vpc)
+        return result
```

Comment:
here is a little workaround for the api "/instances/%s/vpcs"  not returning the description for the vpc. Would be helpful if this api would return the description.

Reply:
yes, that would be great and save some http requests. Thanks

Reply:
I'll see if I can get some time today to test it out